stubs/python-xlib/Xlib/protocol/rq.pyi

```diff
@@ -0,0 +1,416 @@
+        self, data: SliceableBuffer, display: Unused, length: int | Literal["odd", "even"], format: Unused
+    ) -> tuple[tuple[Any, ...], SliceableBuffer]: ...
+
+class List(ValueField):
+    structcode: None
+    type: Struct | ScalarObj | ResourceObj | ClassInfoClass | type[ValueField]
+    pad: int
+    def __init__(
+        self, name: str, type: Struct | ScalarObj | ResourceObj | ClassInfoClass | Type[ValueField], pad: int = ...
+    ) -> None: ...
+    def parse_binary_value(
+        self, data: SliceableBuffer, display: display.Display | None, length: SupportsIndex | None, format: Unused
+    ) -> tuple[list[DictWrapper | None], SliceableBuffer]: ...
+    def pack_value(  # type: ignore[override]  # Override Callable
+        self, val: Sequence[object] | dict[str, object]
+    ) -> tuple[bytes, int, None]: ...
+
+class FixedList(List):
+    size: int
+    def __init__(self, name: str, size: int, type: Struct | ScalarObj, pad: int = ...) -> None: ...
+    def parse_binary_value(
+        self, data: SliceableBuffer, display: display.Display | None, length: Unused, format: Unused
+    ) -> tuple[list[DictWrapper | None], SliceableBuffer]: ...
+
+class Object(ValueField):
+    type: Struct
+    structcode: str | None
+    def __init__(self, name: str, type: Struct, default: int | None = ...) -> None: ...
+    def parse_binary_value(
+        self, data: SliceableBuffer, display: display.Display | None, length: Unused, format: Unused
+    ) -> tuple[DictWrapper, SliceableBuffer]: ...
+    def parse_value(self, val: SliceableBuffer, display: display.Display | None) -> DictWrapper: ...  # type: ignore[override]
+    def pack_value(  # type: ignore[override]  # Override Callable
+        self, val: tuple[object, ...] | dict[str, object] | DictWrapper
+    ) -> bytes: ...
+    def check_value(self, val: tuple[_T, ...] | dict[str, _T] | DictWrapper) -> list[_T]: ...  # type: ignore[override]
+
+class PropertyData(ValueField):
+    structcode: None
+    def parse_binary_value(
+        self, data: SliceableBuffer, display: Unused, length: _IntNew | None, format: int
+    ) -> tuple[tuple[int, SliceableBuffer] | None, SliceableBuffer]: ...
+    def pack_value(  # type: ignore[override]  # Override Callable
+        self, value: tuple[int, Sequence[float] | Sequence[str]]
+    ) -> tuple[bytes, int, Literal[8, 16, 32]]: ...
+
+class FixedPropertyData(PropertyData):
+    size: int
+    def __init__(self, name: str, size: int) -> None: ...
+
+class ValueList(Field):
+    structcode: None
+    keyword_args: int
+    default: str  # type: ignore[assignment]  # Actually different from base class
+    maskcode: bytes
+    maskcodelen: int
+    fields: list[tuple[Field, int]]
+    def __init__(self, name: str, mask: int, pad: int, *fields: Field) -> None: ...
+    def pack_value(  # type: ignore[override]  # Override Callable
+        self, arg: str | dict[str, object], keys: dict[str, object]
```

Comment:
`dict` invariance makes `dict[T, object]` a very annoying annotation for function or method parameters

```suggestion
        self, arg: str | dict[str, Any], keys: dict[str, Any]
```